Typescript types for it.nock

nathanjhaveri opened the pull request:
Types are needed to inform tsc that the jest `it` function has `it.nock`

nathanjhaveri committed:
Typescript types for it.nock

Types are needed to inform tsc that the jest `it` function has `it.nock`

basilenouvellet commented on index.d.ts:
```diff
@@ -0,0 +1,20 @@
+    nockOptions?: nock.Options;
+};
+
+export function upgradeJasmine(global: NodeJS.Global, options?: Options): void;
```
should we do the same for the `upgradeCircus` method?

```suggestion
export function upgradeJasmine(global: NodeJS.Global, options?: Options): void;
export function upgradeCircus(global: NodeJS.Global, options?: Options): void;
```

nathanjhaveri committed:
Update index.d.ts

Co-authored-by: Basile Nouvellet <b.nouvellet@gmail.com>

saerdnaer commented:
@kommander Does anything speak against merging this?

bramski commented:
The file will need to be added to the `package.json` as `types` in order to be consumed.

bramski commented:
I'm very much a fan here so we don't have to ts-ignore anywhere we use this or add a hacky type declaration.

nathanjhaveri committed:
Fix global type that was removed from @types/node

This jest issue has more context https://github.com/facebook/jest/issues/11640

NodeJS.Global was removed from node types in version 16.  Typescript has
introducted `globalThis` and we can use `typeof globalThis` to reference
the global scope. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis
